File: backend/src/main/java/com/kiszka/prj/entities/Message.java
package com.kiszka.prj.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "messages")
@NoArgsConstructor @Data
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "message_id")
    private int id;
    @Column(nullable = false)
    private String senderType;
    @Column(nullable = false)
    private int senderId;
    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;
    @Column(nullable = false)
    private LocalDateTime sentAt = LocalDateTime.now();
}

File: frontend/src/types/message.ts
export type MessageReceived = {
  id: string;
  senderType: string;
  senderId: number;
  content: string;
  sentAt: string;
}

export type MessageToSent = {
  senderType: string;
  content: string;
}

export type PeopleInfo = {
  id: number;
  name: string;
  type: string;
}